src/adapter/src/coord/read_policy.rs

```diff
@@ -272,6 +275,94 @@ impl crate::coord::Coordinator {
             .set_read_policy(storage_policy_updates);
     }
 
+    // If there is not capability for the given object, initialize one at the
+    // earliest possible since. Return the capability.
+    //
+    // When a `compaction_window` is given, this is installed as the policy of
+    // the collection, regardless if a capability existed before or not.
+    fn ensure_compute_capability(
+        &mut self,
+        instance_id: &ComputeInstanceId,
+        id: &GlobalId,
+        compaction_window: Option<CompactionWindow>,
+    ) -> &mut ReadCapability<mz_repr::Timestamp> {
+        let entry = self
+            .compute_read_capabilities
+            .entry(*id)
+            .and_modify(|capability| {
+                // If we explicitly got a compaction window, override any existing
+                // one.
+                if let Some(compaction_window) = compaction_window {
+                    capability.base_policy = compaction_window.into();
+                }
+            })
+            .or_insert_with(|| {
+                let policy: ReadPolicy<Timestamp> = match compaction_window {
+                    Some(compaction_window) => compaction_window.into(),
+                    None => {
+                        // We didn't get an initial policy, so set the current
+                        // since as a static policy.
+                        let compute = self.controller.active_compute();
+                        let collection = compute
+                            .collection(*instance_id, *id)
+                            .expect("collection does not exist");
+                        let read_frontier = collection.read_capability().clone();
+                        ReadPolicy::NoPolicy {
+                            initial_since: read_frontier,
+                        }
```

Comment:
It's complicated: sometimes it's used as a "I don't know, but keep the since at this while we figure it out" policy. But you're right, I will change it to just use `ValidFrom` here.